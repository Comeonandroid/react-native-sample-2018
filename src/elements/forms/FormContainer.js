import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { View } from '../index'

const FormContainer = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

FormContainer.propTypes = {
  children: PropTypes.node
}

const Element = styled(View) `
  padding: 0px 12px;
`

export default FormContainer