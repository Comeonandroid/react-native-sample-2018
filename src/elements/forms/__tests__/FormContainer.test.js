import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import FormContainer from '../FormContainer'

describe('<FormContainer />', () => {
  itBehavesLike('aThemedComponent', { subject: <FormContainer /> })
  itBehavesLike('aComponentWithChildren', { subject: <FormContainer /> })
})
