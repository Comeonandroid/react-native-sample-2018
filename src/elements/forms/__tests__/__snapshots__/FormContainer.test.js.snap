// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormContainer /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<FormContainer /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<FormContainer /> renders component when passed in 1`] = `
<Styled(View)>
  <Component />
</Styled(View)>
`;

exports[`<FormContainer /> renders text when passed in 1`] = `
<Styled(View)>
  Test
</Styled(View)>
`;
